libs/agno/agno/tools/linkup.py

```diff
@@ -0,0 +1,53 @@
+
+from agno.tools import Toolkit
+from typing import Any, List, Optional
+from os import getenv
+from agno.utils.log import logger
+
+try:
+    from linkup import LinkupClient
+except ImportError as e:
+    raise ImportError(
+        "`linkup-sdk` not installed. Please install using `pip install linkup-sdk`")
+
+class LinkupTools(Toolkit):
+    def __init__(
+        self,
+        search: bool = True,
+        api_key: Optional[str] = None,
+        depth: str = "standard",
+        output_type: str = "searchResults",
+        **kwargs
+    ):
+        self.api_key = api_key or getenv("LINKUP_API_KEY")
+        if not self.api_key:
+            logger.error("LINKUP_API_KEY not set. Please set the LINKUP_API_KEY environment variable.")
+
+        self.linkup = LinkupClient(api_key=api_key)
+        self.depth = depth
+        self.output_type = output_type
+
+        tools: List[Any] = []
+        if search:
+            tools.append(self.web_search_with_linkup)
+
+        super().__init__(name="linkup_tools", tools=tools, **kwargs)
+
+    def web_search_with_linkup(self, query:str,  depth: str = 'standard', output_type: str='searchResults') -> str:
```

Comment:
Here & in the `__init__` function, is there a reason not to use `Literal`s for `depth` and `output_type`?